lib/Service/OutboxService.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Mail App
+ *
+ * @copyright 2022 Anna Larch <anna.larch@gmx.net>
+ *
+ * @author Anna Larch <anna.larch@gmx.net>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Mail\Service;
+
+use OC\EventDispatcher\EventDispatcher;
+use OCA\Mail\Account;
+use OCA\Mail\Contracts\ILocalMailboxService;
+use OCA\Mail\Contracts\IMailManager;
+use OCA\Mail\Contracts\IMailTransmission;
+use OCA\Mail\Db\LocalMessage;
+use OCA\Mail\Db\LocalMessageMapper;
+use OCA\Mail\Db\Recipient;
+use OCA\Mail\Events\OutboxMessageCreatedEvent;
+use OCA\Mail\Service\Attachment\AttachmentService;
+use OCP\AppFramework\Db\DoesNotExistException;
+
+class OutboxService implements ILocalMailboxService {
+
+	/** @var IMailTransmission */
+	private $transmission;
+
+	/** @var LocalMessageMapper */
+	private $mapper;
+
+	/** @var AttachmentService */
+	private $attachmentService;
+
+	/** @var EventDispatcher */
+	private $eventDispatcher;
+
+	/** @var IMailManager */
+	private $mailManager;
+
+	public function __construct(IMailTransmission  $transmission,
+								LocalMessageMapper $mapper,
+								AttachmentService  $attachmentService,
+								EventDispatcher    $eventDispatcher,
+								IMailManager $mailManager) {
+		$this->transmission = $transmission;
+		$this->mapper = $mapper;
+		$this->attachmentService = $attachmentService;
+		$this->eventDispatcher = $eventDispatcher;
+		$this->mailManager = $mailManager;
+	}
+
+	/**
+	 * @param array $recipients
```

Comment:
```suggestion
	 * @param Recipient[] $recipients
```

Reply:
it's not.

Reply:
Ideally this should be pushed up to the controller where we do the conversion of stringly typed HTTP into the PHP world